### PR TITLE
[FLINK-22974][dist] Add execution checkpointing related configs in flink-conf.yaml

### PR DESCRIPTION
**Pull request by tinawenqiao**
### What is the purpose of the change
Add execution checkpointing related configs in flink-conf.yaml. That may cause checkpoint failure according to the current configuration.

### Brief change log
Execution checkpointing configs such as  execution.checkpointing.interval, execution.checkpointing.externalized-checkpoint-retention, execution.checkpointing.mode, etc are added in flink-conf.yaml.

### Verifying this change
This change is a trivial rework  without any test coverage.

### Does this pull request potentially affect one of the following parts:
- Dependencies (does it add or upgrade a dependency): (no)
- The public API, i.e., is any changed class annotated with @Public(Evolving): (no)
- The serializers: (no)
- The runtime per-record code paths (performance sensitive): (no)
- Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
- The S3 file system connector: (no)

### Documentation
Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by Myasuka** [comment]
@tinawenqiao As I said in https://github.com/apache/flink/pull/16147, please modify your description of this PR, it's not recommended to leave the description as the default template.

**Comment by tinawenqiao** [comment]
> @tinawenqiao As I said in #16147, please modify your description of this PR, it's not recommended to leave the description as the default template.

The pr description has been modified.

File: flink-dist/src/main/resources/flink-conf.yaml [hunk 1]
```diff
@@ -107,7 +107,18 @@ parallelism.default: 1
 #==============================================================================
 
 # The backend that will be used to store operator state checkpoints if
-# checkpointing is enabled.
+# checkpointing is enabled. Checkpointing is enabled when execution.checkpointing.interval > 0.
+#
+# Execution checkpointing related parameters.
+#
+# execution.checkpointing.interval: -1
```
Comment (Myasuka) [inline flink-dist/src/main/resources/flink-conf.yaml:114]: I think setting `execution.checkpointing.interval` as `-1` is illegal as minimal checkpoint time is 10ms. Maybe we could set it as `180 s` with some comments.

File: flink-dist/src/main/resources/flink-conf.yaml [hunk 2]
```diff
@@ -107,7 +107,18 @@ parallelism.default: 1
...
+# execution.checkpointing.externalized-checkpoint-retention: [DELETE_ON_CANCELLATION, RETAIN_ON_CANCELLATION]
+# execution.checkpointing.max-concurrent-checkpoints: 1
+# execution.checkpointing.min-pause: 0
+# execution.checkpointing.mode: [EXACTLY_ONCE, AT_LEAST_ONCE]
+# execution.checkpointing.timeout: 600000
```
Comment (Myasuka) [inline flink-dist/src/main/resources/flink-conf.yaml:119]: I think a more readable value `10 min` is better.

**Review by Myasuka — CHANGES_REQUESTED** [review]
Thanks for your update!
I tried to load the configuration and noticed that `execution.checkpointing.interval: -1` is not a good example for out-of-box usage.

File: flink-dist/src/main/resources/flink-conf.yaml (hunk 1 above)
Comment (tinawenqiao) [reply flink-dist/src/main/resources/flink-conf.yaml:114]: In CheckpointConfig, checkpointInterval default value is -1.  -1 means checkpoint is disabled. And I found that default value is 500ms in function enableCheckpointing().  I am confused  which value is more reasonable.

File: flink-dist/src/main/resources/flink-conf.yaml (hunk 2 above)
Comment (tinawenqiao) [reply flink-dist/src/main/resources/flink-conf.yaml:119]: Good idea.

File: flink-dist/src/main/resources/flink-conf.yaml (hunk 1 above)
Comment (Myasuka) [reply flink-dist/src/main/resources/flink-conf.yaml:114]: You could refer to [CheckpointConfig#setCheckpointInterval](https://github.com/apache/flink/blob/60c7d9e77a6e9d82e0feb33f0d8bc263dddf2fd9/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java#L219-L227) for more details.

Comment (Myasuka) [reply flink-dist/src/main/resources/flink-conf.yaml:114]: From my point of view, user might uncomment this line for out-of-box usage. However, the value of `-1` is illegal and what I thought is like
~~~ bash
# The interval in which checkpoints are periodically scheduled.
#
# execution.checkpointing.interval: 3 min
~~~

**Comment by tinawenqiao** [comment]
interval is set to 500ms.

File: flink-dist/src/main/resources/flink-conf.yaml (hunk 1 above)
Comment (tinawenqiao) [reply flink-dist/src/main/resources/flink-conf.yaml]: 500ms or 180s？In order to be consistent with the code, is it more appropriate to use 500ms？

Comment (Myasuka) [reply flink-dist/src/main/resources/flink-conf.yaml]: Actually, the default value of `500ms` you said really confused me, this option is a user-control value and have no default value. Where did you find the `500ms` configuration?

Comment (tinawenqiao) [reply flink-dist/src/main/resources/flink-conf.yaml]: You can see it here [StreamExecutionEnvironment#enableCheckpointing](https://github.com/apache/flink/blob/60c7d9e77a6e9d82e0feb33f0d8bc263dddf2fd9/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java/#L518)

Comment (Myasuka) [reply flink-dist/src/main/resources/flink-conf.yaml]: This method is already deprecated, and `500ms` is really too quick for production users.

Comment (tinawenqiao) [reply flink-dist/src/main/resources/flink-conf.yaml]: Yes, I also think it's too quick. I will change it to 3min.

**Review by Myasuka — APPROVED** [review]
The broken CI has no relationship with this PR. LGTM

**Pull request merged**